pkg/api/file_repository.go

```diff
@@ -79,15 +79,20 @@ func (r *FileSystemRepository) FindByName(name string) (*Definition, error) {
 	return definition, nil
 }
 
-// FindByListenPath searches an existing Proxy definition by its listen_path
-func (r *FileSystemRepository) FindByListenPath(path string) (*Definition, error) {
+// Exists searches an existing Proxy definition by its listen_path
+func (r *FileSystemRepository) Exists(def *Definition) (bool, error) {
+	_, ok := r.definitions[def.Name]
```

Comment:
Reading from maps is not thread safe, we should use `sync.RWMutex` `RLock()` method